openhtf/__init__.py

```diff
@@ -161,14 +162,27 @@ def plug_type_map(self):
     return plug_type_map
 
 
-def SetupFramework():
-  """Sets up various bits of the framework. Only needs to be called once."""
-  try:
-    sys.argv = FLAGS(sys.argv)  # parse flags
-  except gflags.FlagsError, e:  # pylint: disable=invalid-name
-    print '%s\nUsage: %s ARGS\n%s' % (e, sys.argv[0], FLAGS)
-    sys.exit(1)
+def CreateArgParser():
+  """Creates an argparse.ArgumentParser for parsing command line flags."""
+  parser = argparse.ArgumentParser('OpenHTF-based testing')
+  # openhtf.io.user_input
+  # Option 1:
+  parser.add_argument('--prompt_timeout_s', type=int,
+                      help='User prompt timeout in seconds.')
+  # Option 2:
+  parser = argparse.ArgumentParser('OpenHTF-based testing', parents=[
```

Comment:
hm, should you also include the conf parser in the parents?  while we don't expect this parser to actually parse those args, without it the config flags won't show up in the help string... and those parent parsers should specify `add_help=False`, according to https://docs.python.org/2.7/library/argparse.html#parents